ccl_chromium_indexeddb.py

```diff
@@ -660,15 +666,15 @@ def get_blob_info(self, db_id: int, store_id: int, raw_key: bytes, file_index: i
         else:
             raise KeyError((db_id, store_id, raw_key, file_index))
 
-    def get_blob(self, db_id: int, store_id: int, raw_key: bytes, file_index: int) -> typing.BinaryIO:
+    def get_blob(self, db_id: int, store_id: int, raw_key: bytes, file_index: int, data_path: os.PathLike) -> typing.BinaryIO:
         # Some detail here: https://github.com/chromium/chromium/blob/master/content/browser/indexed_db/docs/README.md
         if self._blob_dir is None:
             raise ValueError("Can't resolve blob if blob dir is not set")
         info = self.get_blob_info(db_id, store_id, raw_key, file_index)
 
         # path will be: origin.blob/database id/top 16 bits of blob number with two digits/blob number
         # TODO: check if this is still the case on non-windows systems
-        path = pathlib.Path(self._blob_dir, str(db_id), f"{info.blob_number >> 8:02x}", f"{info.blob_number:x}")
+        path = self._blob_dir.joinpath(data_path)
```

Comment:
on second thought, it's probably easier to delete the `data_path` variable above & revert most of the changes to this code.

i'd tried making some changes to `data_path`, when it seemed like i was missing IndexedDB blobs files. those changes didn't have any effect until i changed things around like this. but i see now that the `data_path` doesn't seem to be used; deleting it would alleviate potential confusion.

Reply:
I'm not sure this overload is a good idea. I take your point about not recalculating the path, but unless you know how to generate the path, it should always be done using the `IndexedDBExternalObject` and other parameters, so I'm not sure where else you would ever want to explicitly pass a path, which might open it up to misuse.

Reply:
I have eliminated the overload. I also took to heart the idea that path generatino should be done using the `IndexDBExternalObject` and other parameters, and made common getters for these components.

More so than avoiding recalculating the path, the far more useful option of not recalculating the info became an option when reading blobs.